README.md

```diff
@@ -12,31 +12,39 @@ __work in progress__
 
 (is being tested on digitalocean ubuntu 16.04)
 
-## Ubuntu ##
+## Remote Linux via SSH ##
 
-Create a server (e.g., digitalocean on some other place) with Ubuntu 16.04 on it. 
+Create a server (e.g., digitalocean on some other place) with one of the following tested OS on it:
+* Ubuntu 17.10 x64
+* Ubuntu 16.04 x64
+* Debian 8.10 x64
 
 Clone this repository on your local machine.
 
 Now run: 
 
 ```
-ssh root@[ip exit node] 'bash -s' < create_exitnode.sh [ip exit node]
+exit_node_ip=165.227.241.194
+./build/remote root@$exit_node_ip
 ```
 
-Expected output should be something like:
+This should SSH to the exit node, configure it, reboot, and wait for the reboot to complete.
+
+## Docker
+
+If you don't have a Linux host around, but have access to docker (e.g. a Mac), you can run an exitnode as a docker container.
 
 ```
-Get:1 http://security.ubuntu.com/ubuntu xenial-security InRelease [102 kB]
-Hit:2 http://ams2.mirrors.digitalocean.com/ubuntu xenial InRelease
-Get:3 http://security.ubuntu.com/ubuntu xenial-security/main Sources [108 kB]
-Get:5 http://security.ubuntu.com/ubuntu xenial-security/restricted Sources [2,116 B]
-[...]
-Cloning into '/opt/exitnode'...
-tunneldigger.service is not a native service, redirecting to systemd-sysv-install
-Executing /lib/systemd/systemd-sysv-install enable tunneldigger
-babeld.service is not a native service, redirecting to systemd-sysv-install
-Executing /lib/systemd/systemd-sysv-install enable babeld
+exit_node_ip=165.227.241.194
+# build with default ip at build time
+docker build -t exitnode --build-arg PUBLIC_IP=$exit_node_ip .
+# build with no ip yet (provide PUBLIC_IP at runtime)
+docker build -t exitnode .
+
+# run
+docker run --privileged -e IP=$exit_node_ip exitnode
+# run bash for debugging
+docker run --privileged -it -e IP=$exit_node_ip exitnode bash
```

Comment:
what is the use case for running an exit node in docker? Does this allow for connecting physical home node to a docker container? If so that would  be awesome! Suggest to document how to do this, if you haven't already.

Reply:
> what is the use case for running an exit node in docker?

* I am willing/eager to host an exitnode on my existing cluster, but don't want to pay more $/month for a dedicated VM if I don't have to. My existing cluster dynamically schedules containers across it, as long as I can build the workload into a docker container.
* Even if I didn't have an existing cluster, if I wanted to run other apps on the same server, I'd want to isolate them in separate linux containers so they can use different system deps, and docker/Dockerfiles are one of the easier ways to build/distribute/run lxc containers.

Does this allow for connecting physical home node to a docker container? If so that would be awesome! Suggest to document how to do this, if you haven't already.
* That's the idea. Will document further when I actually get it working. Next steps
  * [ ] - Determine why `docker run ...` won't work on my mac, but does on Travis-CI
  * [ ] - See if the container works on my k8s cluster, since it works on Travis-CI

Reply:
Thanks for taking the time to elaborate! The docker case sounds really cool - and I'd like to suggest to add it as part of a separate pull request after you get it working and with examples of how to use it. Perhaps @bennlich and @paidforby have some ideas about this.

Reply:
Sounds useful and practical to me. I was hoping to run nginx and host some webpages on my exit node eventually, but I would want to give access to other contributors. I could create separate UNIX users or run a CMS (like wordpress), still I would worry about potential conflicts or unintended side effects. Wrapping the exitnode up in a docker container would put my mind at ease.

Reply:
very much agree with @gobengo and @paidforby about the benefits of docker. I was making an argument for making sure it works / examples before introduce it. Won't block it, but am weary of introducing functionality that can't be tested / used.

Reply:
A dockerized exit node sounds super awesome 👍

Reply:
@bennlich @jhpoelen FWIW I did diagnose and resolve the above issues, and got it running in Docker and in my Kubernetes cluster (container requires --privileged flag, I think because L2TP requires privileged syscalls?, but that's fine). I have not tested the exitnode yet from a home node.

So I have two things to test, and will do so in this order:
1. exitnode in digitalocean as created by ./build/remote
2. exitnode in docker in digitalocean - just to prove docker does{,n't} create problems
3. exitnode in docker in my kubernetes cluster (on AWS FWIW) - I may never test this, or at least if it doesn't work I'm not in a rush to debug this. No one else needs this setup. Though if the mesh need more exitnodes this is the cheapest place for me to run one (.... or a dozen) because it can be packed into my exiting EC2 Instances.